### PR TITLE
README for NIM Model Deployment

### PR DESCRIPTION
**Pull request by Gaurav141199**
README for NIM Model Deployment to include use of Model Catalog and Model-by-reference.

### CODE REVIEW

File: model-deployment/containers/nim/README.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ This Readme walks through how to use NIM - [ Meta-Llama-3-8B-Instruct](https://h
 * [llama3](https://github.com/meta-llama/llama3) from Meta.
 * [NIM](https://catalog.ngc.nvidia.com/orgs/nim/teams/meta/containers/llama3-8b-instruct) by Nvidia
 
+We describe two approaches to create this Model Deployment on OCI:
+* Download Model using API-KEY from NGC Nvidia (described below)
+* Utilising Model Catalog to store Models in OCI, later used for model deployment
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README.md:10]: Add hyperlink here to the readme ./Readme-model-catalog.md

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## Download NIM Container image and upload to OCIR
+* Pull the latest NIM Image to local machine or through NB session. Tag it with desired name.
+    ```bash
+    docker pull nvcr.io/nim/meta/llama3-8b-instruct:latest
+    docker tag nvcr.io/nim/meta/llama3-8b-instruct:latest odsc-nim-llama3:latest 
+    ```
+## OCI Container Registry
+
+Once you built and pushed the NIM container, you can now use the `Bring Your Own Container` Deployment in OCI Data Science to deploy the Llama3 model.
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:22]: No built

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+# Method 1: Export Model to Model Catalog
+
+Follow the steps mentioned [here][https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/model-deployment/containers/llama2/README.md], refer the section One time download to OCI Model Catalog.
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:26]: this is not hyperlink code
[]()

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+# Method 1: Export Model to Model Catalog
+
+Follow the steps mentioned [here][https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/model-deployment/containers/llama2/README.md], refer the section One time download to OCI Model Catalog. 
+
+We would utilise the above created model in the next steps to create the Model Deployment. 
+
+# Method 2: Model-by-reference
+
+Follow the steps to upload your model to Object Storage [here][https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/LLM/llama3.1-8B-deployment-vLLM-container.md]
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:32]: fix hyperlinks

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+* To deploy the model now in the console, navigate to your [OCI Data Science Project](https://cloud.oracle.com/data-science/project)
+    * Select the project created earlier and then select `Model Deployment`
+    * Click on `Create model deployment`
+    * Under `Default configuration` set following custom environment variables
+        * Key: `MODEL_DEPLOY_PREDICT_ENDPOINT`, Value: `/v1/completions`
+        * Key: `MODEL_DEPLOY_HEALTH_ENDPOINT`, Value: `/v1/health/ready`
+        * Key: `NIM_MODEL_NAME`, Value: `/opt/ds/model/deployed_model`
+        * Key: `NIM_SERVER_PORT`, Value `8080`
+        * Key: `SHM_SIZE`, Value: `5g`
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:46]: Did you need it? If not, you can remove

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+* Utilising Model-by-reference Feature to download model
+
+# Pre-requisite
+
+The following are the pre-requisite:
+* Notebook session with internet access (Recommended)
+* Download the Llama 3 8B Instruct Model from [HuggingFace](https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct) or NGC repository.
+
+## Download NIM Container image and upload to OCIR
+* Pull the latest NIM Image to local machine or through NB session. Tag it with desired name.
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:15]: We cant pull image using NB session

Comment (gargnipungarg) [reply model-deployment/containers/nim/README-MODEL-CATALOG.md:15]: call download on workstation

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    docker pull nvcr.io/nim/meta/llama3-8b-instruct:latest
+    docker tag nvcr.io/nim/meta/llama3-8b-instruct:latest odsc-nim-llama3:latest 
+    ```
+## OCI Container Registry
+
+Once NIM container is pushed, you can now use the `Bring Your Own Container` Deployment in OCI Data Science to deploy the Llama3 model.
+
+# Method 1: Export Model to Model Catalog
+
+Follow the steps mentioned [here](https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/model-deployment/containers/llama2/README.md), refer the section One time download to OCI Model Catalog.
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:26]: We discussed adding specific sections Gaurav
You need to add hyperlink to the section, not the whole readme page

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+Follow the steps mentioned [here](https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/model-deployment/containers/llama2/README.md), refer the section One time download to OCI Model Catalog. 
+
+We would utilise the above created model in the next steps to create the Model Deployment. 
+
+# Method 2: Model-by-reference
+
+Follow the steps to upload your model to Object Storage [here](https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/LLM/llama3.1-8B-deployment-vLLM-container.md)
+
+Utilise the [section](https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/LLM/llama3.1-8B-deployment-vLLM-container.md) on Create Model by Reference using ADS to create the model.
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:34]: same here

File: model-deployment/containers/nim/README.md [hunk 9]
```diff
@@ -54,10 +58,10 @@ When experimenting with new frameworks and models, it is highly advisable to att
 
 ## Deploy on OCI Data Science Model Deployment
 
-Once you built and pushed the NIM container, you can now use the `Bring Your Own Container` Deployment in OCI Data Science to deploy the Llama3 model.
+Once NIM container is pushed, you can now use the `Bring Your Own Container` Deployment in OCI Data Science to deploy the Llama3 model.
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README.md:61]: here we are building and pushing
my comment for removing build was only other Readme

File: model-deployment/containers/nim/README-MODEL-CATALOG.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
+# Overview
+
+Utilising Model Catalog to store Models in OCI. We describe two ways to achieve this: 
+
+* Storing zipped model file in Model Catalog
+* Utilising Model-by-reference Feature to download model
```
Comment (gargnipungarg) [inline model-deployment/containers/nim/README-MODEL-CATALOG.md:6]: We cant call it model by reference. External customers cant understand this.
Use Object storage to store the model and creating a model catalog pointing to Object storage bucket

**Review by gargnipungarg — CHANGES_REQUESTED** [review]

**Review by gargnipungarg — APPROVED** [review]

**Review by liudmylaru — APPROVED** [review]

**Pull request merged**